src/components/App/App.js

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import "./App.css";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
+import ProtectedRoute from "../ProtectedRoute";
+import {
+  Route,
+  Switch,
+  withRouter,
+  useHistory
+} from "react-router-dom";
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import Footer from "../Footer/Footer";
+import SavedNews from "../SavedNews/SavedNews";
+import { SavedCardsContext } from "../../contexts/SavedCardsContext";
+import mainApi from "../../utils/MainApi";
+import SignInPopup from "../Popups/SignInPopup";
+import SignUpPopup from "../Popups/SignUpPopup";
+import InfoToolTip from "../Popups/InfoToolTip";
+
+function App({ props }) {
+  const history = useHistory();
+
+  const [currentUser, setCurrentUser] = React.useState({});
+
+  React.useEffect(initLoggedIn, []);
+
+  const [savedCards, setSavedCards] = React.useState([]);
+
+  const [isSignInPopupOpen, setIsSignInPopupOpen] = React.useState(false);
+  const [isSignUpPopupOpen, setIsSignUpPopupOpen] = React.useState(false);
+  const [isInfoToolTipOpen, setIsInfoToolTipOpen] = React.useState(false);
+  const [signUpMessage, setSignUpMessage] = React.useState(null);
+  const [overlayOpen, setOverlayOpen] = React.useState(false);
+
+  function getSavedArticles() {
+    mainApi
+      .getSavedArticles()
+      .then((articles) => {
+        setSavedCards(articles.data);
+      })
+      .catch((err) => console.log(err));
+  }
+
+  function deleteArticle(id) {
+    mainApi
+      .deleteArticle(id)
+      .then(() => setSavedCards(savedCards.filter((c) => c._id !== id)))
+      .catch((error) => console.log(error));
+  }
+
+  function saveArticle(content, input) {
+    mainApi
+      .saveArticle(content, input)
+      .then(() => {
+        getSavedArticles();
+      })
+      .catch((error) => console.log(error));
+  }
+
+  function closeAllPopups() {
+    setIsSignInPopupOpen(false);
+    setIsSignUpPopupOpen(false);
+    setIsInfoToolTipOpen(false);
+  }
+
```

Comment:
**Might be better**

You can add an `Escape` listener like this:

```jsx
 useEffect(() => {
      const closeByEscape = (e) => {
        if (e.key === 'Escape') {
          closeAllPopups();
        }
      }

      document.addEventListener('keydown', closeByEscape)
      
      return () => document.removeEventListener('keydown', closeByEscape)
  }, [])

```

Here you define function `closeByEscape` inside the `useEffect` so that it could maintain a **fixed** reference if the component rerenders.
And of course we need to return a `clean up` function to **`remove`** the listener if the component unmounts.
And don’t forget about an **`empty dependencies array`** in order to add the listener only once.